server/src-lib/Hasura/GraphQL/Execute/RemoteJoins.hs

```diff
@@ -283,14 +272,16 @@ insertBatchResults accumResp Batch{..} remoteResp =
 
     -- splice the remote result into the hasura result with the proper key (and
     -- at the right index), removing phantom fields
+    spliceRemote :: OJ.Value -> OJ.Object -> Either GQJoinError OJ.Object 
     spliceRemote remoteDataVal accumData = do
       peeledValue <- peelOffNestedFields batchNestedFields remoteDataVal
       pure $
-        -- TODO Brandon note:
-        -- document rrPhantomFields/batchPhantoms so that we can understand if or why it's okay that the OJ.delete might silently be a noop here:
+        -- Strip phantom fields
         foldl' (flip OJ.delete)
                 -- TODO Brandon note:
```

Comment:
Good points. Hopefully these will be resolved automatically while sending out the new PRs